src/main/java/seedu/address/ui/ClassDetailCard.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.ui;
+
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.student.Student;
+
+/**
+ * An UI component that displays information of a {@code Student}.
+ */
+public class ClassDetailCard extends UiPart<Region> {
+
+    private static final String FXML = "ClassDetailCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    public final Student student;
+
+    @FXML
+    private HBox classDetail;
+    @FXML
+    private Label name;
+    @FXML
+    private Label classDetails;
+    @FXML
+    private Label attendance;
+    @FXML
+    private Label classParticipation;
+    @FXML
+    private Label assignments;
+
+    /**
+     * Creates a {@code PersonCode} with the given {@code Student} and index to display.
+     */
+    public ClassDetailCard(Student student) {
+        super(FXML);
+        this.student = student;
+        name.setText(student.getName().fullName);
+        classDetails.setText(student.getClassDetails().classDetails);
```

Comment:
Might be good to add a line between the different information so that it is clearer?